github-adapter/src/main/java/be/xplore/githubmetrics/githubadapter/mappingclasses/GHJob.java

```diff
@@ -0,0 +1,48 @@
+package be.xplore.githubmetrics.githubadapter.mappingclasses;
+
+import be.xplore.githubmetrics.domain.domain.Job;
+
+public record GHJob(
+        long id,
+        long run_id,
+        String status,
+        String conclusion
+) {
+
+    Job getJob() {
+
+        return new Job(
+                this.id,
+                this.run_id,
+                convertStatus(),
+                convertConclusion()
+
+        );
+    }
+
+    private Job.JobConclusion convertConclusion() {
+        return switch (this.conclusion) {
+            case "success" -> Job.JobConclusion.SUCCESS;
+            case "failure",
+                    "cancelled",
+                    "timed_out",
+                    "action_required" -> Job.JobConclusion.FAILURE;
+            case "neutral", "skipped" -> Job.JobConclusion.NEUTRAL;
+            case null -> Job.JobConclusion.NEUTRAL;
+            default -> throw new IllegalStateException(
+                    "Unexpected GH Job Conclusion Value: " + this.conclusion
+            );
+        };
+    }
+
+    private Job.JobStatus convertStatus() {
```

Comment:
I feel like this could be part of the JobStatus class. Using some sort of valueOf() implementation could reduce the complexity here